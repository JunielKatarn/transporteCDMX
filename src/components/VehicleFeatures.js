import React, { memo } from 'react'
import SwiperFlatList from 'react-native-swiper-flatlist'
import { Caption, Card, Subheading } from 'react-native-paper'
import { translate } from '../i18n'
import { StyleSheet, View } from 'react-native'
import { sharedStyles } from '../styles'
import { responsiveHeight, responsiveWidth } from 'react-native-responsive-dimensions'

const VehicleFeatures = props => {
  const _goToImpoundLotsMap = () => {
    props.navigation.navigate('Impound Lots Map')
  }

  const _goToVerificentrosMap = () => {
    props.navigation.navigate('Verificentros Map')
  }

  return (
    <SwiperFlatList style={styles.swiperFlatList} pagingEnabled={false}>
      <Card style={styles.card} onPress={_goToImpoundLotsMap}>
        <Card.Cover style={styles.cover} source={{ uri: 'https://picsum.photos/500/300' }} />
        <Card.Content>
          <Subheading style={styles.cardSubheading}>{translate('impound_lots')}</Subheading>
          <Caption>{translate('impound_lots_description')}</Caption>
        </Card.Content>
      </Card>
      <Card style={styles.card} onPress={_goToVerificentrosMap}>
        <Card.Cover style={styles.cover} source={{ uri: 'https://picsum.photos/500/300' }} />
        <Card.Content>
          <Subheading style={styles.cardSubheading}>{translate('verificentros')}</Subheading>
          <Caption>{translate('verificentros_description')}</Caption>
        </Card.Content>
      </Card>
      <View style={styles.phanthomView} />
    </SwiperFlatList>
  )
}

const styles = StyleSheet.create({
  swiperFlatList: {
    ...sharedStyles.mt3,
    ...sharedStyles.pl3,
    ...sharedStyles.pb3,
  },
  card: {
    ...sharedStyles.mr3,
    width: responsiveWidth(80),
    height: responsiveHeight(45),
  },
  cover: {
    height: responsiveHeight(28),
  },
  cardSubheading: {
    ...sharedStyles.mt2,
  },
  phanthomView: {
    width: responsiveWidth(10),
  },
})

export default memo(VehicleFeatures)
