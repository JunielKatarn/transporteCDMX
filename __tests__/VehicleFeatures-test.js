import React from 'react'
import { shallow } from 'enzyme'
import VehicleFeatures from '../src/components/VehicleFeatures'

describe('VehicleFeatures', () => {
  it('it renders correctly', () => {
    const wrapper = shallow(<VehicleFeatures />)
    expect(wrapper).toMatchSnapshot()
  })
})
