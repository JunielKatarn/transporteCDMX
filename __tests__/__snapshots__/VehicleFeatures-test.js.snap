// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleFeatures it renders correctly 1`] = `
<ForwardRef
  PaginationComponent={[Function]}
  autoplay={false}
  autoplayDelay={3}
  autoplayInvertDirection={false}
  autoplayLoop={false}
  data={Array []}
  disableGesture={false}
  index={0}
  pagingEnabled={false}
  renderAll={false}
  showPagination={false}
  style={
    Object {
      "marginTop": 17,
      "paddingLeft": 17,
    }
  }
  vertical={false}
  viewabilityConfig={Object {}}
>
  <withTheme(Card)
    onPress={[Function]}
    style={
      Object {
        "height": 600.3000000000001,
        "marginBottom": 17,
        "marginRight": 17,
        "width": 600,
      }
    }
  >
    <withTheme(Card.Cover)
      source={
        Object {
          "testUri": "../../../src/assets/images/impoundLot.jpg",
        }
      }
      style={
        Object {
          "height": 373.52000000000004,
        }
      }
    />
    <Card.Content>
      <Subheading
        style={
          Object {
            "marginTop": 11.22,
          }
        }
      >
        [missing "en.impound_lots" translation]
      </Subheading>
      <Caption>
        [missing "en.impound_lots_description" translation]
      </Caption>
    </Card.Content>
  </withTheme(Card)>
  <withTheme(Card)
    onPress={[Function]}
    style={
      Object {
        "height": 600.3000000000001,
        "marginBottom": 17,
        "marginRight": 17,
        "width": 600,
      }
    }
  >
    <withTheme(Card.Cover)
      source={
        Object {
          "testUri": "../../../src/assets/images/verificentros.jpg",
        }
      }
      style={
        Object {
          "height": 373.52000000000004,
        }
      }
    />
    <Card.Content>
      <Subheading
        style={
          Object {
            "marginTop": 11.22,
          }
        }
      >
        [missing "en.verificentros" translation]
      </Subheading>
      <Caption>
        [missing "en.verificentros_description" translation]
      </Caption>
    </Card.Content>
  </withTheme(Card)>
  <View
    style={
      Object {
        "width": 75,
      }
    }
  />
</ForwardRef>
`;
